schema/schema_test.go

```diff
@@ -0,0 +1,183 @@
+package schema_test
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/hpcloud/termui/schema"
+
+	term "github.com/cloudfoundry/cli/cf/terminal"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// tests for schema
+func TestSchemaString(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Schema")
+}
+
+var _ = Describe("Schema", func() {
+	Describe("Test if schema asks correctly for a string", func() {
+		Context("ask for string", func() {
+			It("should be Insert string value for /stringtest [required]", func() {
+				ui := term.NewUI(os.Stdin, term.NewTeePrinter())
```

Comment:
Also, I'd feel a lot safer if we pass in `&bytes.Buffer{}` instead of `os.Stdin` (to avoid accidentally blocking on interactive user input).